### PR TITLE
Pack CHANGELOG.md with .nupkg when present, include link to release notes

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Add CHANGELOG.md to Nuget Packages
Add link to releasenotes
Add CHANGELOG.md for Azure.Template

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
The Event Hubs bits look good to me.

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: Didn't you also plan to try and pull out the specific version release notes?

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: I don't remember if we had an issue for this but didn't we also want to include the README.md in the package as well?

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (weshaggard) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: Are you planning to delete this, or are you trying to wait for freedback from keyvault?

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: If you have cleaned out all the existing PackageReleaseNotes properties you should remove this part of the condition and instead and an error if a project sets that property.

**Review by tg-msft — APPROVED** [review]
This is a cool change!

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: I'm releasing a new version today or tomorrow, so if this isn't ready I prefer preview.8 go out as-is.

File: eng/Directory.Build.Data.targets
Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: That will happen in on of the release steps once the changelog is present in the nuget.

File: eng/Directory.Build.Data.targets
Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: I was not aware of that. I can add it here if needed.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (chidozieononiwu) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: I wasn't sure what version those notes were for.

File: eng/Directory.Build.Data.targets
Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: I removed PackageReleaseNotes  for Track 2 libraries. Some track one libraries still specify PackageReleaseNotes while some like servicebus want to use the changelog. I am trying to accommodate both cases.

Comment (weshaggard) [reply eng/Directory.Build.Data.targets]: How are you going to pull out the changelog snippet and put it into the package metadata during the release step? I know we will do that for the github release but I'd expect for the package we would have to do it here somewhere.

Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: Filled Issue to get that done here https://github.com/Azure/azure-sdk-for-net/issues/9179

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [reply eng/Directory.Build.Data.targets]: Yes I think it would be nice to include the README in the package as well in the same fashion.

Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: Included

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: Don't comment those out just yet. I haven't been able to release yet (waiting for sign-off on #9174).

**Review by heaths — CHANGES_REQUESTED** [review]
Don't make changes to certificates\changelog.md just yet. I need to release first.

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (chidozieononiwu) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: Am still testing to see how it all works on nuget. However you can add the notes to the changelog and it should be fine. With a link to CHANGELOG.md from nuget which is how all packages will be till I work on pulling out the appropriate release note.

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: One other thing we need to do is start to run the `eng/pipelines/templates/scripts/replace-relative-links.yml@azure-sdk-tools` step before we produce the packages so that the links in the packages are stable. We probably want to do that for readmes and changelogs. Lets not block this PR on this but can you please file another issue to track that work.

**Review by weshaggard — APPROVED** [review]

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/Azure.Security.KeyVault.Certificates.csproj]: I've released preview 8, so you can just remove this section now.

**Review by heaths — APPROVED** [review]

File: eng/Directory.Build.Data.targets
Comment (chidozieononiwu) [reply eng/Directory.Build.Data.targets]: Issue filled here https://github.com/Azure/azure-sdk-for-net/issues/9210

**Pull request merged**
